src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalOptionsPanel.java

```diff
@@ -126,6 +126,7 @@ public void save(SonarLintGlobalSettings settings) {
     getService(CleanAsYouCodeService.class).setFocusOnNewCode(focusOnNewCode.isSelected());
     settings.setAutoTrigger(autoTrigger.isSelected());
     settings.setNodejsPath(nodeJsPath.getText());
+    settings.setFocusOnNewCode(focusOnNewCode.isSelected());
```

Comment:
This is normally done through line 126, but the problem is that the Settings instance used in the CaYCService is not right. So good catch!
I suggest keeping this responsibility in the service, but passing the settings instance to the setFocusOnNewCode()